tiledb/common/scoped_executor.h

```diff
@@ -74,6 +74,19 @@ class ScopedExecutor final {
   DISABLE_COPY_AND_COPY_ASSIGN(ScopedExecutor);
   DISABLE_MOVE_ASSIGN(ScopedExecutor);
 
+  /**
+   * Helper method to allow setting/changing function as needed.
+   * This can be used to change/disable a function should a routine
+   * need to disable or modify a prior function that may have been
+   * intended for premature/error exit cleanup and would cause problems
+   * if exited at some point later than appropriate.
+   *
+   * @param fn the function to install
+   */
+  void set_fn(std::function<void()>&& fn) {
```

Comment:
Please don't make this change. `ScopedExecutor` is already hackish enough as it is, and this would make it worse.